### PR TITLE
Add parameter checks to all keepers with tests

### LINKED ISSUE #536: Add parameter checks to all keepers with tests
See #490 and #532 for the list of all parameters

### PR DESCRIPTION
**Pull request by prin-r**
Fix #536

### CODE REVIEW

File: chain/x/zoracle/exec_env_test.go [hunk 1]
```diff
@@ -167,6 +169,45 @@ func TestRequestExternalData(t *testing.T) {
...
+	envErr = env.RequestExternalData(1, 42, []byte("prepare32"))
+	require.Nil(t, envErr)
+	envErr = env.RequestExternalData(1, 43, []byte("prepare32"))
+	require.Nil(t, envErr)
+	envErr = env.RequestExternalData(1, 44, []byte("prepare32"))
+	require.Nil(t, envErr)
+	envErr = env.RequestExternalData(1, 45, []byte("prepare32"))
+	require.Nil(t, envErr)
+
+	// Should got an error if trying to request more external data
```
Comment (sorawit) [inline chain/x/zoracle/exec_env_test.go]: Should get

**Review by sorawit — DISMISSED** [review]
Looks good to me. @taobun please review

File: chain/x/zoracle/internal/keeper/raw_request.go [hunk 2]
```diff
@@ -59,7 +62,12 @@ func (k Keeper) AddNewRawDataRequest(
 		// TODO: fix error later
 		return types.ErrRequestNotFound(types.DefaultCodespace)
 	}
-	k.SetRawDataRequest(ctx, requestID, externalID, types.NewRawDataRequest(dataSourceID, calldata))
+
+	err := k.SetRawDataRequest(ctx, requestID, externalID, types.NewRawDataRequest(dataSourceID, calldata))
```
Comment (taobun) [inline chain/x/zoracle/internal/keeper/raw_request.go]: It's my fault, please change `SetRawDataRequest` signature to not return an error and you don't need to check error when `SetRawDataRequest`

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]
lgtm

**Review by sorawit — APPROVED** [review]

**Pull request merged**